lib/prawn/markup/processor/text.rb

```diff
@@ -78,8 +93,9 @@ def end_sup
         append_text('</sup>')
       end
 
-      def start_color
-        rgb, c, m, y, k = current_attrs.values_at('rgb', 'c', 'm', 'y', 'k')
+      def start_color(options = {})
+        options = current_attrs unless options.any?
+        rgb, c, m, y, k = options.transform_keys(&:to_s).values_at('rgb', 'c', 'm', 'y', 'k')
```

Comment:
Options will have to be adjusted to support either string or array. Probably it's easier to extract a new method that does the `append_text` part based on the string/array color and leaves the current_attrs parsing to `start_color`.

Reply:
All `start_` and `end_` Methods are callbacks for the SAX processor. Coloring links is not XML-initiated but an internal operation. Hence I suggest to create a method `append_color_tag(color)` that checks if `color` is an array and appends the color tag with CMYK attributes or with an RGB attribute otherwise. The `start_color` method can then parse `current_attrs` into either a CMYK array or a RGB value and call `append_color_tag` with that, while `start_a` calls `append_color_tag(link_options[:color])`.